Added Kyverno policies and README.md

mzeevi opened the pull request:
Added different policies to the `policies/` directory and made some changes to the existing ones. All the policies are also documented in the `README.md` file.

Signed-off-by: mzeevi <meytar80@gmail.com>

mzeevi commented:
Test on a OpenShift cluster to see if everything works. I only tested what I could on `KinD`

cc @sahar2339 @nadavdolin

mzeevi commented:
Test on a OpenShift cluster to see if everything works. I only tested what I could on KinD

/cc @sahar2339 @nadavdolin

nadavdolin commented on policies/deny-configmaps.yml:
```diff
@@ -0,0 +1,38 @@
+          all:
+          - key: "{{ privileged.data.groups | parse_yaml(@) }}"
+            operator: AllNotIn
+            value: "{{ request.userInfo.groups }}"
```
add condition also for privileged username

nadavdolin commented on policies/deny-techpreviewupgrade.yml:
```diff
@@ -0,0 +1,22 @@
+    validate:
+      message: Managed Runb Kubernetes Service (RKS) Customers may not use TechPreviewNoUpgrade FeatureGate that could prevent any future ability to do an upgrade of their clusters.
+      deny:
+        conditions:
```
Add condition to allow privilege users

nadavdolin commented:
Looks good to me /approve

mzeevi committed:
Added Kyverno policies and README.md
Added different policies to the `policies/` directory and made some changes to the existing ones.
All the policies are also documented in the `README.md` file.

Signed-off-by: mzeevi <meytar80@gmail.com>

mzeevi committed:
Merge branch 'main' into kyverno

nadavdolin commented:
/approve
/lgtm
